File: Part1_강의자료/Part1_쪽지시험/문제별코드/ProblemB_숫자야구/2503.py
N = int(input())
q = [list(map(int, input().split())) for _ in range(N)]

ans = 0
for x in range(1, 10):
    for y in range(1, 10):
        for z in range(1, 10):
            if x == y or y == z or x == z:
                continue
            psb = True
            for i in range(N):
                qx = q[i][0] // 100
                qy = (q[i][0] // 10) % 10
                qz = q[i][0] % 10
                
                strike, ball = 0, 0
                if x == qx:
                    strike += 1
                elif x == qy or x == qz:
                    ball += 1
                if y == qy:
                    strike += 1
                elif y == qx or y == qz:
                    ball += 1
                if z == qz:
                    strike += 1
                elif z == qx or z == qy:
                    ball += 1
                if strike != q[i][1] or ball != q[i][2]:
                    psb = False
                    break
            if psb:
                ans += 1

print(ans)

File: Part1_강의자료/Part1_쪽지시험/문제별코드/ProblemB_숫자야구/2503.cpp
#include <stdio.h>
#include <vector>
using namespace std;

struct Query {
    int num, strike, ball;
};

int main()
{
    int N;
    scanf("%d", &N);

    vector<Query> vec(N);
    for (int i = 0; i < N;i++)
        scanf("%d %d %d", &vec[i].num, &vec[i].strike, &vec[i].ball);

    int ans = 0;
    for (int x = 1; x <= 9; x++)
    {
        for (int y = 1; y <= 9; y++)
        {
            for (int z = 1; z <= 9; z++)
            {
                if (x == y || y == z || x == z)
                    continue;
                bool psb = true;
                for (int i = 0; i < N; i++)
                {
                    int qx = vec[i].num / 100;
                    int qy = (vec[i].num / 10) % 10;
                    int qz = vec[i].num % 10;

                    int strike = 0;
                    int ball = 0;
                    if (x == qx)
                        strike++;
                    else if (x == qy || x == qz)
                        ball++;
                    if (y == qy)
                        strike++;
                    else if (y == qx || y == qz)
                        ball++;
                    if (z == qz)
                        strike++;
                    else if (z == qx || z == qy)
                        ball++;
                    if (strike != vec[i].strike || ball != vec[i].ball)
                    {
                        psb = false;
                        break;
                    }
                }
                if (psb)
                    ans++;
            }
        }
    }

    printf("%d\n", ans);
    return 0;
}
